open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,736 @@
+                    ConsentExtensionConstants.AMOUNT);
+            if (!(Boolean.parseBoolean(maximumIndividualAmountResult.
+                    getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                return maximumIndividualAmountResult;
+            }
+        } else {
+            log.error(ErrorConstants.MISSING_MAXIMUM_INDIVIDUAL_AMOUNT);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.MISSING_MAXIMUM_INDIVIDUAL_AMOUNT);
+
+        }
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Validates the Currency in Maximum Individual Amount in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response.
+     */
+    public static JSONObject validateMaximumIndividualAmountCurrency(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        Object maximumIndividualAmount = controlParameters.
+                get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT);
+
+
+        JSONObject maximumIndividualAmountResult = validateJsonObjectKey((JSONObject) maximumIndividualAmount,
+                ConsentExtensionConstants.CURRENCY);
+        if (!(Boolean.parseBoolean(maximumIndividualAmountResult.getAsString(ConsentExtensionConstants.IS_VALID)))) {
+            return maximumIndividualAmountResult;
+        }
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Method to validate variable recurring payment periodic limits.
+     * This method validates the periodic limits specified in the control parameters for variable recurring payments.
+     * It checks if the provided JSON array of periodic limits is valid and then iterates through each limit
+     * to ensure that required fields such as amount, currency, period alignment, and period type are present and
+     * meet the specified criteria.
+     *
+     * @param controlParameters Initiation Object containing periodic limits
+     * @return validation response object indicating whether the provided periodic limits are valid
+     */
+    public static JSONObject validatePeriodicLimits(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        // Check if the periodic limits key is present
+        if (controlParameters.containsKey(ConsentExtensionConstants.PERIODIC_LIMITS)) {
+
+            // Retrieve the periodic limits from the control parameters
+            Object periodicLimit = controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+            String errorMessage = String.format(ErrorConstants.INVALID_PARAMETER_MESSAGE, "periodic limit",
```

Comment:
Have we used this `errorMessage ` variable anywhere?

Reply:
Fixed in [e8d8048]